### PR TITLE
Draggable scrollbar

### PR DESCRIPTION
**Pull request by Helguli**
A fix to issue #619.
Everything looks the same as before, but now you can drag the scrollbar thumb to navigate in the editor.
I can make a settings entry for it too, if it's necessary.

### CODE REVIEW

**Comment by gsantner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/net/gsantner/markor/ui/DraggableScrollbarScrollView.java  9
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/gsantner/markor/pullRequest?prid=5568816&bid=18262492)

File: CONTRIBUTORS.md [hunk 1]
```diff
@@ -43,3 +43,4 @@ Where:
 * **[David Hebbeker](https://david.hebbeker.info/)**<br/>~° Added tooltips for text actions.
 * **[Harshad Srinivasan](https://github.com/harshad1)**<br/>~° Added functionality to handle leading spaces in lists.
 * **[Niels](https://stackoverflow.com/users/432115/niels)**<br>~° Added encryption for textfile contents.
+* **[Helguli](helguli@yahoo.com)**<br>~° Added fast scroll to editor.
```
Comment (gsantner) [inline CONTRIBUTORS.md:46]: I suggest you to use a webpage unless you like spam. Just a hint.

File: app/src/main/java/net/gsantner/markor/ui/DraggableScrollbarScrollView.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        setSmoothScrollingEnabled(true);
+        setLtr();
+    }
+
+    public DraggableScrollbarScrollView(Context context, AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+        setSmoothScrollingEnabled(true);
+        setLtr();
+    }
+
```
Comment (gsantner) [inline app/src/main/java/net/gsantner/markor/ui/DraggableScrollbarScrollView.java:33]: Combine the 2/3 constructors above to just use the `this()` constructor for the remaining one

**Comment by Helguli** [comment]
> What do you think, does it also make sense to add it to View-Mode?

IMO, it does make sense, but since the View-Mode is in a WebView (and not ScrollView), I haven't tried it yet.

File: app/src/main/java/net/gsantner/markor/ui/DraggableScrollbarScrollView.java (hunk 2 above)
Comment (Helguli) [reply app/src/main/java/net/gsantner/markor/ui/DraggableScrollbarScrollView.java:33]: That resulted in odd behavior, and after noticing that inside the constructors, getLayoutDirection() always returns LTR, I moved that call to onAttachedToWindow().

**Comment by gsantner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/net/gsantner/markor/ui/DraggableScrollbarScrollView.java  9
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/gsantner/markor/pullRequest?prid=5568816&bid=18262492)

File: app/src/main/java/net/gsantner/markor/ui/DraggableScrollbarScrollView.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        super(context);
+    }
+
+    public DraggableScrollbarScrollView(Context context, AttributeSet attrs) {
+        super(context, attrs);
+    }
+
+    public DraggableScrollbarScrollView(Context context, AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+    }
```
Comment (gsantner) [inline app/src/main/java/net/gsantner/markor/ui/DraggableScrollbarScrollView.java:26]: so custom steps are not required anymore at constructor, do I see right?

In that case it should be possible to remove all constructors...and super's contructors (`Scrollview()`) are automatically called

**Comment by gsantner** [comment]
Found it to work. Would be cool to have this on View mode/WebView as well. However let's have it in an own PR.

**Comment by Helguli** [comment]
You're welcome @gsantner 
I'll create a new branch and a PR as I start working on the View mode.

**Comment by harshad1** [comment]
Hi. I can't get this to work. Do I need to enable something somewhere? I see the scrollbar when I flick-scroll, but I cannot grab it.

**Comment by Helguli** [comment]
@harshad1 
Hi. You don't need to enable anything. It is a bit hard to grab, being at the side of the screen.

**Pull request merged**